### ISSUE TITLE
Request: Automatically configure compiler to reflect the test module's pom file (or build.gradle)

### ISSUE THREAD

**Issue by rsmckinney** [google/compile-testing#346]
Most non-trivial projects will have a test module configured with dependencies, compiler options, processor path, etc. I just want the compile-testing stuff to reflect the build's configuration. Otherwise, getting this right, and keeping it in sync is tedious and error prone.  Would greatly appreciate a feature to do this automatically.

**Comment by eamonnmcmanus** [comment]
I understand the desire for this, but I think it would be hard to achieve within the `compile-testing` library itself. It seems as if it should be a Maven or Gradle plugin, or the like?

**Comment by rsmckinney** [comment]
Yes, that makes perfect sense. Ideally, this feature would be based on separate plugins targeting maven and gradle. In my case I just need maven support.

Perhaps a crazy idea. Suppose compile-testing could provide a Java compiler plugin that captures the build configuration exactly as reflected in the compiler during compilation of compile-testing tests. The information could be collected and serialized in a form directly usable during execution of compile-testing tests.
